server/potentialMatch.py

```diff
@@ -0,0 +1,84 @@
+import mysql.connector
+from mysql.connector import errorcode
+from connect import connectToDB
+
+def findPotentialMatches(id):
+    try:
+        connection = connectToDB()
+        if(connection != False):
+            cursor = connection.cursor(buffered=True)
+
+            genderPreference = getGenderPreference(id, cursor, connection)
+            if genderPreference == "":
+                return {"response": "No Matches", "id": id, "Number of Matches": 0}
+
+            potentialIds = getPotentialIds(id, genderPreference, cursor, connection)
+            if len(potentialIds) == 0:
```

Comment:
isn't it possible to write a query that checks their gender preference and use that to get potential ids instead of doing separate queries

Reply:
I need to do a conditional based on the genderPreference because the query is different if they selected "Both" vs. "Female/Male"